lib/src/oauth2.dart

```diff
@@ -84,16 +84,17 @@ void logout(SystemCache cache) {
   }
 }
 
-/// Asynchronously passes an OAuth2 [Client] to [fn], and closes the client when
-/// the [Future] returned by [fn] completes.
+/// Asynchronously passes an OAuth2 [Client] to [fn].
+///
+/// Does not close the client, since that would close the shared client. It must
+/// be closed elsewhere.
 ///
 /// This takes care of loading and saving the client's credentials, as well as
 /// prompting the user for their authorization. It will also re-authorize and
 /// re-run [fn] if a recoverable authorization error is detected.
 Future<T> withClient<T>(SystemCache cache, Future<T> Function(Client) fn) {
   return _getClient(cache).then((client) {
     return fn(client).whenComplete(() {
-      client.close();
```

Comment:
It would be better to just use a new client... or to wrap all access to the other client with a  `withClient` function in `http.dart`.

Reply:
Added a TODO.
All client.close() does is closing the shared (inner) client. And that gets closed in the PubCommand.run.